monkey/infection_monkey/telemetry/messengers/batching_telemetry_messenger.py

```diff
@@ -0,0 +1,94 @@
+import queue
+import threading
+import time
+from typing import Dict
+
+from infection_monkey.telemetry.i_batchable_telem import IBatchableTelem
+from infection_monkey.telemetry.i_telem import ITelem
+from infection_monkey.telemetry.messengers.i_telemetry_messenger import ITelemetryMessenger
+
+DEFAULT_PERIOD = 5
+WAKES_PER_PERIOD = 4
+
+
+class BatchingTelemetryMessenger(ITelemetryMessenger):
```

Comment:
Since the responsibility of this class is not to send the telemetries but to batch them. I wonder if we should make a distinction in the name, because a decorator now has exatcly the same naming convention as a messanger even though the responsibility and creation of the object is different

Reply:
We could name it `BatchingTelemetryMessengerDecorator`.